src/machines/modelingMachine.ts

```diff
@@ -261,1088 +264,544 @@ export const getPersistedContext = (): Partial<PersistedModelingContext> => {
+      if (trap(info)) return false
+      return info.enabled
+    },
+    'Can constrain horizontally align': ({ context: { selectionRanges } }) => {
+      const info = horzVertDistanceInfo({
+        selectionRanges,
+        constraint: 'setHorzDistance',
+      })
+      if (trap(info)) return false
+      return info.enabled
+    },
+    'Can constrain vertically align': ({ context: { selectionRanges } }) => {
+      const info = horzVertDistanceInfo({
+        selectionRanges,
+        constraint: 'setHorzDistance',
+      })
+      if (trap(info)) return false
+      return info.enabled
+    },
+    'Can constrain snap to X': ({ context: { selectionRanges } }) => {
+      const info = absDistanceInfo({
+        selectionRanges,
+        constraint: 'snapToXAxis',
+      })
+      if (trap(info)) return false
+      return info.enabled
+    },
+    'Can constrain snap to Y': ({ context: { selectionRanges } }) => {
+      const info = absDistanceInfo({
+        selectionRanges,
+        constraint: 'snapToYAxis',
+      })
+      if (trap(info)) return false
+      return info.enabled
+    },
+    'Can constrain equal length': ({ context: { selectionRanges } }) => {
+      const info = setEqualLengthInfo({ selectionRanges })
+      if (trap(info)) return false
+      return info.enabled
+    },
+    'Can canstrain parallel': ({ context: { selectionRanges } }) => {
+      const info = equalAngleInfo({ selectionRanges })
+      if (err(info)) return false
+      return info.enabled
+    },
+    'Can constrain remove constraints': ({
+      context: { selectionRanges },
+      event,
+    }) => {
+      const _event = event as Extract<
+        ModelingMachineEvent,
+        { type: 'Constrain remove constraints' }
+      >
+      const info = removeConstrainingValuesInfo({
+        selectionRanges,
+        pathToNodes: _event.data && [_event.data],
+      })
+      if (trap(info)) return false
+      return info.enabled
+    },
+    'Can convert to variable': ({ event }) => {
+      const _event = event as Extract<
+        ModelingMachineEvent,
+        { type: 'Convert to variable' }
+      >
+      if (!_event.data) return false
+      const ast = parse(recast(kclManager.ast))
+      if (err(ast)) return false
+      const isSafeRetVal = isNodeSafeToReplacePath(ast, _event.data.pathToNode)
+      if (err(isSafeRetVal)) return false
+      return isSafeRetVal.isSafe
+    },
+    'next is tangential arc': ({ context: { sketchDetails } }) => {
+      return (
+        (state?.event as any).data.tool === 'tangentialArc' &&
+        isEditingExistingSketch({ sketchDetails })
+      )
+    },
+    'next is rectangle': ({ context: { sketchDetails } }) => {
+      if ((state?.event as any).data.tool !== 'rectangle') return false
+      return canRectangleTool({ sketchDetails })
+    },
+    'next is line': () => (state?.event as any).data.tool === 'line',
+    'next is none': () => (state?.event as any).data.tool === 'none',
```

Comment:
In XState v5, `state.event` is removed. Another way to approach this is to store the **mode** in the context and reason based on that in these guards. cc @franknoirot

Reply:
I believe I've resolved this by using context to store the `currentTool` and `currentMode` of the modeling machine.